pipelinewise/fastsync/partialsync/utils.py

```diff
@@ -0,0 +1,63 @@
+import os
+import re
+
+from pipelinewise.fastsync.commons import utils
+
+
+def upload_to_s3(snowflake, file_parts, temp_dir):
+    """Upload exported data into S3"""
+
+    s3_keys = []
+    for file_part in file_parts:
+        s3_keys.append(snowflake.upload_to_s3(file_part, tmp_dir=temp_dir))
+        os.remove(file_part)
+
+    # Create a pattern that match all file parts by removing multipart suffix
+    s3_key_pattern = (
+        re.sub(r'\.part\d*$', '', s3_keys[0])
+        if len(s3_keys) > 0
+        else 'NO_FILES_TO_LOAD'
+    )
+    return s3_keys, s3_key_pattern
+
+
+def load_into_snowflake(snowflake, args, s3_keys, s3_key_pattern, size_bytes):
+    """load data into Snowflake"""
+
+    # delete partial data from the table
+    target_schema = utils.get_target_schema(args.target, args.table)
+    table_dict = utils.tablename_to_dict(args.table)
+    target_table = table_dict.get('table_name')
+    where_clause = f'WHERE {args.column} >= {args.start_value}'
+    if args.end_value:
+        where_clause += f' AND {args.column} <= {args.end_value}'
+
+    snowflake.query(f'DELETE FROM {target_schema}."{target_table.upper()}" {where_clause}')
```

Comment:
In full fastsync, it creates a temp table and then swap it with the original table. here in partial we agreed to delete that part of data from the table. otherwise data will be duplicated if there is some data for that part. 
by deleting we can be sure the data won't be duplicated and will be updated if the source is changed.